lib/companies/pending_changes.ex

```diff
@@ -71,6 +71,19 @@ defmodule Companies.PendingChanges do
     end
   end
 
+  def get_pending_changes_for(record) do
+    resource_name = struct_to_string(record)
+    id = record.id |> to_string()
+
+    query =
+      from p in PendingChange,
+        where: p.resource == ^resource_name,
+        where: is_nil(p.approved),
+        where: fragment("changes->>'id' = ?", ^id)
+
+    Repo.all(query)
+  end
```

Comment:
The use case for this function is to determine if any pending changes exist. The place that calls this function isn't really interested in the actual list of changes. What would you think about converting this to a [predicate function](https://hexdocs.pm/elixir/naming-conventions.html#trailing-question-mark-foo)?

```ex
def any?(record) do
  # ...

  Repo.exists?(query)
end
```

`Companies.JobChecker.check_job/1` could then simplify its conditional like this:
```ex
if PendingChanges.any?(record) do 
  Jobs.delete(job, %{id: user_id})
end
```